agents/ten_packages/system/ten_ai_base/interface/ten_ai_base/llm.py

```diff
@@ -0,0 +1,119 @@
+#
+# This file is part of TEN Framework, an open source project.
+# Licensed under the Apache License, Version 2.0.
+# See the LICENSE file for more information.
+#
+from abc import ABC, abstractmethod
+import asyncio
+from enum import Enum
+from typing import Optional, TypedDict, Union
+
+from pydantic import BaseModel
+from ten import (
+    AsyncExtension,
+    TenEnv,
+    Data,
+)
+from ten.async_ten_env import AsyncTenEnv
+from ten.cmd import Cmd
+from ten.cmd_result import CmdResult, StatusCode
+from .const import CMD_PROPERTY_TOOL, CMD_TOOL_REGISTER
+from .types import TenLLMCallCompletionArgs, TenLLMDataCompletionArgs, TenLLMToolMetadata
+from .helper import AsyncQueue
+import json
+
+class TenLLMBaseExtension(AsyncExtension, ABC):
```

Comment:
Add the comment for what exactly will be abstracted in this class?
1. Tool Registry?
2. Queue Item?
3. Flush?